Services/UI/OJS.Services.Ui.Business/Implementations/ContestsBusinessService.cs

```diff
@@ -91,34 +91,46 @@ public async Task<ContestDetailsServiceModel> GetContestDetails(int id)
             var contestActivityEntity = await this.activityService
                 .GetContestActivity(contestDetailsServiceModel!.Map<ContestForActivityServiceModel>());
 
-            var participant = await this.participantsData
+            contestDetailsServiceModel!.CanBeCompeted = contestActivityEntity.CanBeCompeted;
+            contestDetailsServiceModel.CanBePracticed = contestActivityEntity.CanBePracticed;
+            contestDetailsServiceModel.IsAdminOrLecturerInContest = isLecturerInContestOrAdmin;
+
+            var competeParticipant = await this.participantsData
                 .GetWithProblemsForParticipantsByContestByUserAndIsOfficial(
                     id,
                     user.Id,
-                    contestActivityEntity.CanBeCompeted);
+                    true);
 
-            contestDetailsServiceModel!.CanBeCompeted = contestActivityEntity.CanBeCompeted;
-            contestDetailsServiceModel.CanBePracticed = contestActivityEntity.CanBePracticed;
-            contestDetailsServiceModel.IsAdminOrLecturerInContest = isLecturerOrAdmin;
+            var participantToGetProblemsFrom = contestDetailsServiceModel.CanBeCompeted
+                ? competeParticipant
+                : await this.participantsData.GetWithProblemsForParticipantsByContestByUserAndIsOfficial(
+                    id,
+                    user.Id,
+                    false);
```

Comment:
Why did you split this into 2 variables? In the end is always a single participant that is used. Just use `contestActivityEntity.CanBeCompeted` for the variable. Also the same boolena can be used where you check `competeParticipant == null`. If the contest can be competed by the participant, this means compete participant must be present, or I'm missing something.

Reply:
In the details page the problems that are shown are based on the participant. I split the participant variable so i can determine if the user can view compete or practice results separately.

Reply:
So why not use the activityService.CanBeCompeted for this?